src/js/components/FocusedContainer.js

```diff
@@ -44,24 +123,31 @@ export const FocusedContainer = ({
   }, [bodyOverflowStyle, hidden, trapFocus, restrictScroll]);
 
   useEffect(() => {
+    const roots = contextRoots.current;
     const timer = setTimeout(() => {
-      if (!hidden && trapFocus && roots && roots[0]) {
-        roots.forEach(makeNodeUnfocusable);
+      if (!hidden && trapFocus && roots?.[0]) {
+        // make every root before this one unfocusable
+        roots.forEach((root, index) => {
+          if (index < roots.length - 1) makeNodeUnfocusable(root);
+        });
       }
     }, 0);
 
-    return () => {
-      // remove trap and restore ability to focus on the last root only
-      if (roots && roots[0]) makeNodeFocusable(roots[roots.length - 1]);
-      clearTimeout(timer);
-    };
-  }, [hidden, roots, trapFocus]);
+    return () => clearTimeout(timer);
```

Comment:
within here do we need to make the nodesFocusable again? 
above we have 
```
  // make every root before this one unfocusable
        roots.forEach((root, index) => {
          if (index < roots.length - 1) makeNodeUnfocusable(root);
        });

```

so in the return do we need 

      // Make all roots focusable again
      roots.forEach((root) => {
        makeNodeFocusable(root);
      });

Reply:
I've done it in the other useEffect once we splice the roots array to the updated array on unmount.